#warning that have not been squelched bypass CI

For example https://github.com/RobotLocomotion/drake/pull/14135 introduced a `#warning` for deprecation purposes, but that warning shows up when building Drake from source, not just downstream when used in external projects:
```
[7:05:21 AM]  INFO: From Compiling systems/analysis/simulator_flags.cc:
[7:05:21 AM]  In file included from systems/analysis/simulator_flags.cc:1:
[7:05:21 AM]  bazel-out/k8-opt/bin/systems/analysis/_virtual_includes/simulator_flags/drake/systems/analysis/simulator_flags.h:3:2: warning: "This file has been merged into simulator_config_functions.h. This file and build target will be removed from Drake on or after 2021-01-01."  // NOLINT(whitespace/line_length) [-W#warnings]
[7:05:21 AM]  #warning "This file has been merged into simulator_config_functions.h. This file and build target will be removed from Drake on or after 2021-01-01."  // NOLINT(whitespace/line_length)
[7:05:21 AM]   ^
[7:05:21 AM]  1 warning generated.
```
https://github.com/RobotLocomotion/drake/pull/14149 fixes the issue for this particular `#warning`, but CI/bazel should (presumably) catch these warnings and prevent them from being merged to master.

--- a/tools/skylark/drake_cc.bzl
+++ b/tools/skylark/drake_cc.bzl
@@ -6,6 +6,7 @@ load("@cc//:compiler.bzl", "COMPILER_ID", "COMPILER_VERSION_MAJOR")
 # building with any compiler.
 CXX_FLAGS = [
     "-Werror=all",
+    "-Werror=cpp",
     "-Werror=deprecated",
     "-Werror=deprecated-declarations",
     "-Werror=ignored-qualifiers",

